builtins/amp-ad.js

```diff
@@ -284,6 +285,28 @@ export function installAd(win) {
       return loadPromise(this.iframe_);
     }
 
+    /** @override  */
+    unlayoutCallback() {
+      if (this.iframe_) {
+        removeElement(this.iframe_);
+        if (this.placeholder_) {
+          this.togglePlaceholder(true);
+        }
+        if (this.fallback_) {
+          this.toggleFallback(false);
+        }
+
+        this.iframe_ = null;
+        this.intersectionObserver_ = null;
```

Comment:
It does also listen on viewport. But it looks like all listeners are freed via `onViewportCallback(false)` which is called from the element's `onViewportCallback`. Could you confirm if you always call `onViewportCallback(false)` on the element before unlayout?